### ISSUE TITLE
[Troubleshoot]: Rollout time

### ISSUE THREAD

**Issue by lozoRUST** [RLinf/RLinf#587]
### Problem description

I'm trying the training of openvla-oft on libero-90, using the configuration file libero_90_openvlaoft.yaml. The log demonstrates that the rollout is extremely slow. I wonder whether it is normal? By the way, I was running the scripts with 4 A100 80GB.

<img width="1050" height="146" alt="Image" src="https://github.com/user-attachments/assets/530cd5a8-19af-4f25-84b9-7ae7732e1187" />

### Configuration YAML file

YAML config file:

You can also paste the full config here.
```
defaults:
  - env/libero_90@env.train
  - env/libero_90@env.eval
  - model/openvla_oft@actor.model
  - training_backend/fsdp@actor.fsdp_config
  - override hydra/job_logging: stdout

hydra:
  run:
    dir: .
  output_subdir: null
  searchpath:
    - file://${oc.env:EMBODIED_PATH}/config/

cluster:
  num_nodes: 1
  component_placement:
    actor,env,rollout: all

runner:
  task_type: embodied
  logger:
    log_path: "../results"
    project_name: rlinf
    experiment_name: "libero_90_grpo_openvlaoft"
    logger_backends: ["tensorboard"] # wandb, swanlab

  max_epochs: 1000
  max_steps: -1

  only_eval: False
  val_check_interval: -1
  save_interval: 25

  resume_dir: null # Optional: path to a saved checkpoint directory, such as 'checkpoints/global_step_10'. If not None, it will be used to resume training.
  ckpt_path: null  # Optional: path to a .pt checkpoint. If not None, it will be loaded after the model is instantiated (for evaluation).

algorithm:
  normalize_advantages: True
  kl_penalty: kl  # how to estimate kl divergence: kl or kl_penalty
  group_size: 8
  rollout_epoch: 64
  eval_rollout_epoch: 1 # set eval_rollout_epoch > 0 when enable runner.only_eval or runner.val_check_interval > 0

  reward_type: action_level  # action_level or chunk_level
  logprob_type: token_level
  entropy_type: token_level

  adv_type: grpo
  loss_type: actor
  loss_agg_func: "token-mean"
  kl_beta: 0.0
  entropy_bonus: 0
  clip_ratio_high: 0.28
  clip_ratio_low: 0.2
  clip_ratio_c: 3.0
  value_clip: 0.2
  huber_delta: 10.0

  gamma: 0.99
  gae_lambda: 0.95

  reward_coef: 5.0

  sampling_params:
    do_sample: True
    temperature_train: 1.6
    temperature_eval: 1.6
    top_k: -1
    top_p: 1.0
    repetition_penalty: 1.0

  # length argument for autoregressive sampling
  # max length means max amount of tokens to generate
  length_params:
    max_new_token: null
    max_length: 1024
    min_length: 1

  filter_rewards: True
  rewards_lower_bound: 0.5
  rewards_upper_bound: 4.5

env:
  group_name: "EnvGroup"
  enable_offload: False

  # Override the default values in env/libero_90
  train:
    total_num_envs: 64
    max_episode_steps: 512 # max episode steps for truncation
    max_steps_per_rollout_epoch: 512
    reward_coef: ${algorithm.reward_coef}
    group_size: ${algorithm.group_size}
    reset_gripper_open: False
  eval:
    total_num_envs: 500
    auto_reset: True
    ignore_terminations: True
    max_episode_steps: 512
    max_steps_per_rollout_epoch: 512
    group_size: 1
    use_fixed_reset_state_ids: True
    use_ordered_reset_state_ids: True
    reset_gripper_open: False
    is_eval: True
    video_cfg:
      save_video: True
      video_base_dir: ${runner.logger.log_path}/video/eval

rollout:
  group_name: "RolloutGroup"
  generation_backend: "huggingface"
  enable_offload: True
  pipeline_stage_num: 1

  model:
    model_path: "/root/workspace/RLinf/RLinf-OpenVLAOFT-LIBERO-90-Base-Lora"
    precision: ${actor.model.precision}

actor:
  group_name: "ActorGroup"
  training_backend: "fsdp"
  micro_batch_size: 32
  global_batch_size: 16384
  seed: 1234
  enable_offload: True

  # Override the default values in model/openvla_oft
  model:
    model_path: "/root/workspace/RLinf/RLinf-OpenVLAOFT-LIBERO-90-Base-Lora"
    model_type: "openvla_oft"
    unnorm_key: libero_90_no_noops_trajall
    max_prompt_length: 128
    
  optim:
    lr: 2.0e-5
    value_lr: 3.0e-3
    adam_beta1: 0.9
    adam_beta2: 0.999
    adam_eps: 1.0e-05
    weight_decay: 0.01
    clip_grad: 1.0

  # Override the default values in training_backend/fsdp
  fsdp_config:
    strategy: "fsdp"
    gradient_checkpointing: True
    mixed_precision:
      param_dtype: ${actor.model.precision}
      reduce_dtype: ${actor.model.precision}
      buffer_dtype: ${actor.model.precision}

reward:
  use_reward_model: False

critic:
  use_critic_model: False
```

### Log file

You can find the log file in `logs/` folder or the $output_dir/$experiment_name folder (defined in the yaml config) if you are using our example scripts.

Log file:

If you cannot find the log, please provide the full log messages here.
```
<paste log here>
```

### Environment

python -V:
Python 3.11.14
uv pip list:
Using Python 3.11.14 environment at: /root/workspace/RLinf/.venv
Package                            Version     Editable project location
---------------------------------- ----------- ----------------------------------
absl-py                            2.3.1
accelerate                         1.12.0
addict                             2.4.0
aiohappyeyeballs                   2.6.1
aiohttp                            3.13.3
aiohttp-cors                       0.8.1
aiosignal                          1.4.0
annotated-doc                      0.0.4
annotated-types                    0.7.0
antlr4-python3-runtime             4.11.1
anyio                              4.12.1
argcomplete                        3.6.3
arm-pytorch-utilities              0.4.3
array-record                       0.8.3
asttokens                          3.0.1
astunparse                         1.6.3
attrs                              25.4.0
av                                 16.1.0
bddl                               3.6.0
blinker                            1.9.0
boto                               2.49.0
cachetools                         5.5.2
certifi                            2026.1.4
cffi                               2.0.0
cfgv                               3.5.0
charset-normalizer                 3.4.4
click                              8.3.1
cloudpickle                        3.1.2
colorama                           0.4.6
colorful                           0.6.0a1
configargparse                     1.7.1
contourpy                          1.3.3
crcmod                             1.7
cryptography                       43.0.3
cycler                             0.12.1
dacite                             1.9.2
dash                               4.0.0rc6
datasets                           3.6.0
decorator                          5.2.1
deprecated                         1.3.1
diffusers                          0.30.3
dill                               0.3.8
distlib                            0.4.0
distro                             1.9.0
dlimp                              0.0.1
dm-tree                            0.1.9
docstring-parser                   0.17.0
donfig                             0.8.1.post1
draccus                            0.8.0
easydict                           1.13
einops                             0.8.1
etils                              1.13.0
evdev                              1.9.2
executing                          2.2.1
farama-notifications               0.0.4
fast-kinematics                    0.2.2
fastapi                            0.128.0
fasteners                          0.20
fastjsonschema                     2.21.2
filelock                           3.20.3
flash-attn                         2.7.4.post1
flask                              3.1.2
flatbuffers                        25.12.19
fonttools                          4.61.1
frozenlist                         1.8.0
fsspec                             2025.3.0
future                             1.0.0
gast                               0.7.0
gcs-oauth2-boto-plugin             3.3
gitdb                              4.0.12
gitpython                          3.1.46
glfw                               2.10.0
google-api-core                    2.29.0
google-apitools                    0.5.35
google-auth                        2.39.0
google-auth-httplib2               0.3.0
google-auth-oauthlib               1.2.3
google-crc32c                      1.8.0
google-pasta                       0.2.0
google-reauth                      0.1.1
googleapis-common-protos           1.72.0
grpcio                             1.76.0
gsutil                             5.35
gym                                0.26.2
gym-notices                        0.1.0
gymnasium                          0.29.1
h11                                0.16.0
h5py                               3.15.1
hf-xet                             1.2.1rc0
httpcore                           1.0.9
httplib2                           0.20.4
httpx                              0.28.1
huggingface-hub                    0.36.0
hydra-core                         1.4.0.dev1
icmplib                            3.0.4
identify                           2.6.16
idna                               3.11
imageio                            2.37.2
imageio-ffmpeg                     0.6.0
immutabledict                      4.2.2
importlib-metadata                 8.0.0
importlib-resources                6.5.2
iniconfig                          2.3.0
ipython                            9.9.0
ipython-pygments-lexers            1.1.1
itsdangerous                       2.2.0
jedi                               0.19.2
jinja2                             3.1.6
jiter                              0.12.0
joblib                             1.5.3
json-numpy                         2.1.1
jsonlines                          4.0.0
jsonschema                         4.26.0
jsonschema-specifications          2025.9.1
jupyter-core                       5.9.1
jupytext                           1.19.0.dev0
keras                              2.15.0
kiwisolver                         1.4.10rc0
latex2sympy2                       1.9.0
libclang                           18.1.1
libero                             0.1.0       /root/workspace/RLinf/.venv/libero
liger-kernel                       0.6.4
llvmlite                           0.44.0
lxml                               6.0.2
mani-skill                         3.0.0b22
markdown                           3.10
markdown-it-py                     4.0.0
markupsafe                         3.0.3
matplotlib                         3.10.8
matplotlib-inline                  0.2.1
mdit-py-plugins                    0.5.0
mdurl                              0.1.2
mergedeep                          1.3.4
ml-dtypes                          0.2.0
monotonic                          1.6
mplib                              0.1.1
mpmath                             1.3.0
msgpack                            1.1.2
mujoco                             3.4.0
multidict                          6.7.0
multiprocess                       0.70.16
mypy-extensions                    1.1.0
namex                              0.1.0
narwhals                           2.15.0
nbformat                           5.10.4
nest-asyncio                       1.6.0
networkx                           3.6.1
ninja                              1.13.0
nltk                               3.9.2
nodeenv                            1.10.0
numba                              0.61.2
numcodecs                          0.16.5
numpy                              1.26.4
nvidia-cublas-cu12                 12.4.5.8
nvidia-cuda-cupti-cu12             12.4.127
nvidia-cuda-nvrtc-cu12             12.4.127
nvidia-cuda-runtime-cu12           12.4.127
nvidia-cudnn-cu12                  9.1.0.70
nvidia-cufft-cu12                  11.2.1.3
nvidia-curand-cu12                 10.3.5.147
nvidia-cusolver-cu12               11.6.1.9
nvidia-cusparse-cu12               12.3.1.170
nvidia-cusparselt-cu12             0.6.2
nvidia-ml-py                       13.580.82
nvidia-nccl-cu12                   2.21.5
nvidia-nvjitlink-cu12              12.4.127
nvidia-nvtx-cu12                   12.4.127
nvitop                             1.6.1
oauth2client                       4.1.3
oauthlib                           3.3.1
omegaconf                          2.4.0.dev4
open3d                             0.19.0
openai                             2.15.0
opencensus                         0.11.4
opencensus-context                 0.2.dev0
opencv-python                      4.12.0.88
openexr                            3.4.4
opentelemetry-api                  1.26.0
opentelemetry-exporter-prometheus  0.47b0
opentelemetry-proto                1.39.1
opentelemetry-sdk                  1.26.0
opentelemetry-semantic-conventions 0.47b0
openvla-oft                        0.0.1
opt-einsum                         3.4.0
optree                             0.18.0
orjson                             3.11.5
packaging                          25.0
pandas                             3.0.0rc2
parso                              0.8.5
peft                               0.11.1
pexpect                            4.9.0
pillow                             12.1.0
pip                                25.3
platformdirs                       4.5.1
plotly                             6.5.2
pluggy                             1.6.0
pre-commit                         4.5.1
prettytable                        3.17.0
prometheus-client                  0.24.1
promise                            2.3
prompt-toolkit                     3.0.52
propcache                          0.4.1
proto-plus                         1.27.0
protobuf                           4.25.8
psutil                             7.2.1
ptyprocess                         0.7.0
pure-eval                          0.2.3
py-spy                             0.4.1
pyarrow                            22.0.0
pyasn1                             0.6.1
pyasn1-modules                     0.4.2
pybind11                           3.0.1
pycparser                          2.23
pydantic                           2.12.5
pydantic-core                      2.41.5
pyecharts                          2.0.9
pygments                           2.19.2
pylatexenc                         3.0a33
pynput                             1.8.1
pyopengl                           3.1.10
pyopenssl                          24.2.1
pyparsing                          3.3.1
pyperclip                          1.11.0
pyquaternion                       0.9.9
pyrealsense2                       2.56.5.9235
pytest                             9.0.2
python-dateutil                    2.9.0.post0
python-xlib                        0.33
pytorch-kinematics                 0.7.6
pytorch-seed                       0.2.0
pyu2f                              0.1.5
pyvers                             0.1.0
pyyaml                             6.0.3
pyyaml-include                     1.4.1
ray                                2.47.1
referencing                        0.37.0
regex                              2026.1.15
requests                           2.32.5
requests-oauthlib                  2.0.0
retry-decorator                    2.0a1
retrying                           1.4.2
rich                               13.9.4
robosuite                          1.4.1
rpds-py                            0.30.0
rsa                                4.7.2
ruff                               0.14.11
safetensors                        0.7.0
sapien                             3.0.1
scikit-learn                       1.8.0
scipy                              1.17.0
sentencepiece                      0.1.99
sentry-sdk                         3.0.0a7
setuptools                         75.8.2
simple-parsing                     0.1.7
simplejson                         3.20.2
six                                1.17.0
smart-open                         7.5.0
smmap                              5.0.2
sniffio                            1.3.1
stack-data                         0.6.3
starlette                          0.50.0
swanlab                            0.7.6
sympy                              1.13.1
tabulate                           0.9.0
tensorboard                        2.15.2
tensorboard-data-server            0.7.2
tensordict                         0.10.0
tensorflow                         2.15.0
tensorflow-addons                  0.23.0
tensorflow-datasets                4.9.3
tensorflow-estimator               2.15.0
tensorflow-graphics                2021.12.3
tensorflow-io-gcs-filesystem       0.37.1
tensorflow-metadata                1.17.3
termcolor                          3.3.0
threadpoolctl                      3.6.0
timm                               0.9.10
tokenizers                         0.19.1
toml                               0.10.2
toppra                             0.6.3
torch                              2.6.0
torch-memory-saver                 0.0.9
torchaudio                         2.6.0
torchdata                          0.11.0
torchvision                        0.21.0
tqdm                               4.67.1
traitlets                          5.14.3
transformers                       4.40.1
transforms3d                       0.4.2
trimesh                            4.11.0
triton                             3.2.0
typeguard                          2.13.3
typing-extensions                  4.15.0
typing-inspect                     0.9.0
typing-inspection                  0.4.2
tyro                               1.0.5
urllib3                            2.6.3
uvicorn                            0.40.0
virtualenv                         20.36.1
wandb                              0.24.0
wcwidth                            0.2.14
werkzeug                           3.1.5
wheel                              0.45.1
word2number                        1.1
wrapt                              1.14.2
xxhash                             3.6.0
yarl                               1.22.0
zarr                               3.1.5
nvidia-smi:
Sun Jan 18 20:40:05 2026       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.161.08             Driver Version: 535.161.08   CUDA Version: 12.4     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA A100-SXM4-80GB          On  | 00000000:A1:00.0 Off |                    0 |
| N/A   33C    P0              64W / 400W |      0MiB / 81920MiB |      0%      Default |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|  No running processes found                                                           |
+---------------------------------------------------------------------------------------+
RLinf version:
Docker image tag: 

### Before submitting a new issue...

- [x] Have you checked relevant issues, FAQs (https://rlinf.readthedocs.io/en/latest/rst_source/faq.html), or asked the chatbot at the top right corner of the [documentation page](https://rlinf.readthedocs.io).

**Comment by lozoRUST** [comment]
actually I've changed the rollout epoch to 2. But considering the 1000 max steps, it requires nearly 90 hours to finish the RL training. I wonder if there should be any other places to modify in the yaml file.

**Comment by lozoRUST** [comment]
<img width="2000" height="227" alt="Image" src="https://github.com/user-attachments/assets/10e6ab2e-cc68-4806-a061-b729c1d5b5ae" />
this is much slower than i assumed, I wonder know if that is normal

**Comment by Infinity4B** [comment]
Same problem in libero_10_grpo_openvlaoft. Have you suceeded in libero-10 training?

**Comment by lozoRUST** [comment]
> Same problem in libero_10_grpo_openvlaoft. Have you suceeded in libero-10 training?

not yet, still help wanted. I don't know whether it's a normal training time. But at least it's still running.

**Comment by zoeyuchao** [comment]
We usually use 8*A100 80GB to run experiments. and do you use EGL or omesa?
